common/djangoapps/philu_commands/management/commands/tests/test_mailchimp_commands.py

```diff
@@ -0,0 +1,135 @@
+
+
+class MailChimpUser(TestCase):
+    """
+        Tests for both `sync_users_with_mailchimp` and `delete_users_from_mailchimp` command.
+    """
+
+    @mute_signals(post_save)
+    def setUp(self):
+        super(MailChimpUser, self).setUp()
+        self.user = UserFactory.create(
+            username="test",
+            password="test123",
+            email="test@example.com"
+        )
+        org = OrganizationFactory()
+        org.org_type = "New"
+        org.save()
+        self.user.extended_profile.organization = org
+        self.user.extended_profile.save()
+
+    @mock.patch('mailchimp_pipeline.client.ChimpClient.delete_user_from_list')
+    def test_delete_users_from_mailchimp(self, mock_func):
+        call_command('delete_users_from_mailchimp')
+        mock_func.assert_called_once_with(settings.MAILCHIMP_LEARNERS_LIST_ID, self.user.email)
+
+    @mock.patch('mailchimp_pipeline.client.ChimpClient.add_list_members_in_batch')
+    @mock.patch('django.db.connection')
+    def test_sync_users_with_mailchimp_command(self, mocked_connection, mock_func):
+        users = list(User.objects.all())
+        users_data = self.get_users_data_to_send(users)
+        call_command('sync_users_with_mailchimp')
+        mock_func.assert_called_once_with(settings.MAILCHIMP_LEARNERS_LIST_ID, {"members": users_data})
+
+    @mock.patch('lms.djangoapps.certificates.api.get_certificates_for_user')
+    @mock.patch('mailchimp_pipeline.client.ChimpClient.add_list_members_in_batch')
+    @mock.patch('django.db.connection')
+    def test_sync_users_with_mailchimp_command_for_cert_exception(self, mocked_connection, mock_func,
+                                                                  mocked_get_user_cert_func):
+        mocked_get_user_cert_func.side_effect = Exception
+        users = list(User.objects.all())
+        users_data = self.get_users_data_to_send(users)
+        call_command('sync_users_with_mailchimp')
+        mock_func.assert_called_once_with(settings.MAILCHIMP_LEARNERS_LIST_ID, {"members": users_data})
+        self.assertRaises(Exception)
+
+    @mock.patch('django.db.connection')
+    def test_sync_users_with_mailchimp_command_for_exception(self, mocked_connection):
+        users = list(User.objects.all())
+        users_data = self.get_users_data_to_send(users)
+        call_command('sync_users_with_mailchimp')
+        self.assertRaises(Exception)
+
+    @mock.patch('philu_commands.management.commands.sync_users_with_mailchimp.get_user_active_enrollements')
+    @mock.patch('django.db.connection')
+    def test_sync_users_with_mailchimp_command_for_user_exception(self, mocked_connection, mocked_user_enrollment_func):
+        mocked_user_enrollment_func.side_effect = Exception
+        users = list(User.objects.all())
+        users_data = self.get_users_data_to_send(users)
+        call_command('sync_users_with_mailchimp')
+        self.assertRaises(Exception)
+
+    def get_users_data_to_send(self, users):
```

Comment:
Try to import this function instead of copy/pasting.

Reply:
Done